realm/realm-library/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -241,42 +220,14 @@ public void executeTransaction(Transaction transaction) {
         return RealmResults.createFromDynamicTableOrView(this, tableView, className);
     }
 
-    private static DynamicRealm create(RealmConfiguration configuration) {
-        synchronized (BaseRealm.class) {
-
-            // Check if a cached instance already exists for this thread
-            String canonicalPath = configuration.getPath();
-            Map<RealmConfiguration, Integer> localRefCount = referenceCount.get();
-            Integer references = localRefCount.get(configuration);
-            if (references == null) {
-                references = 0;
-            }
-            Map<RealmConfiguration, DynamicRealm> realms = realmsCache.get();
-            DynamicRealm realm = realms.get(configuration);
-            if (realm != null) {
-                localRefCount.put(configuration, references + 1);
-                return realm;
-            }
-
-            // Create new DynamicRealm and cache it. All exception code paths must close the DynamicRealm otherwise we risk
-            // serving faulty cache data.
-            validateAgainstExistingConfigurations(configuration);
-            boolean autoRefresh = Looper.myLooper() != null;
-            realm = new DynamicRealm(configuration, autoRefresh);
-            List<RealmConfiguration> pathConfigurationCache = globalPathConfigurationCache.get(canonicalPath);
-            if (pathConfigurationCache == null) {
-                pathConfigurationCache = new CopyOnWriteArrayList<RealmConfiguration>();
-                globalPathConfigurationCache.put(canonicalPath, pathConfigurationCache);
-            }
-            pathConfigurationCache.add(configuration);
-            realms.put(configuration, realm);
-            localRefCount.put(configuration, references + 1);
-
-            // Increment global reference counter
-            realm.acquireFileReference(configuration);
-
-            return realm;
-        }
+    /**
+     * Creates a {@link DynamicRealm} instance without checking the existence in the {@link RealmCache}.
+     *
+     * @return a {@link DynamicRealm} instance.
+     */
+    static DynamicRealm createInstance(RealmConfiguration configuration) {
```

Comment:
this will be called in RealmCache when there is no instance in current thread, call this method and put it into the cache. The ideal way could be 1. Ask from cache 2. no finding in cache. 3. Create the instance in `DynamicRealm.java` 4. put the new instance into the cache. Maybe this is better? maybe i can split the `createRealmOrGetFromCache` into two functions, and always create the instance here. But in that way, the problem is, the lock needs to be in this function as well :(